### PR TITLE
Fix for Native Eligibility in Stage

### PR DESCRIPTION
**Pull request by spencersablan**
### Description

We added a [fix for Venmo in stage](#546) last week, but that had some negative effects on the PayPal button. So we want to keep the environment check in, but don't check the environment for Venmo.

### Why are we making these changes? Include references to any related Jira tasks or GitHub Issues
https://paypal.slack.com/archives/CLKPA7686/p1677501252842249

### Reproduction Steps (if applicable)
- In stage
- **Click the paypal button**
- Before this fix: its broken & loads infinitely 
- After this fix: it should work as anticipated

### Groups who should review (if applicable)

@elizabethmv @gregjopa 

❤️  Thank you!

### CODE REVIEW

**Review by wsbrunson — APPROVED** [review]

**Review by elizabethmv — APPROVED** [review]

File: src/payment-flows/native/eligibility.js [hunk 1]
```diff
@@ -134,10 +134,12 @@ export function canUseNativeQRCode({ fundingSource, win } : {| fundingSource : ?
 }
 
 export function isNativeEligible({ props, config, serviceData } : IsEligibleOptions) : boolean {
-    const { fundingSource, onShippingChange, createBillingAgreement, createSubscription, platform } = props;
+    const { fundingSource, onShippingChange, createBillingAgreement, createSubscription, env, platform } = props;
     const { firebase: firebaseConfig } = config;
     const { cookies, merchantID, fundingEligibility } = serviceData;
     const isVenmoEligible = fundingEligibility?.venmo?.eligible;
+    const isVenmoButton = fundingSource === FUNDING.VENMO;
```
Comment (gregjopa) [inline src/payment-flows/native/eligibility.js:141]: Does this `isNativeEligible()` function get executed on button click or on render?

I'm wondering if `fundingSource` will be defined for the smart stack or only for a standalone venmo button.

Comment (spencersablan) [reply src/payment-flows/native/eligibility.js:141]: Both actually! It is called on render & on button click.

Comment (gregjopa) [reply src/payment-flows/native/eligibility.js:141]: Are you able to test out the smart stack with this change to ensure Venmo still works as expected?

**Review by gregjopa — APPROVED** [review]

File: src/payment-flows/native/eligibility.js (hunk 1 above)
Comment (spencersablan) [reply src/payment-flows/native/eligibility.js:141]: My comments didn't post in this thread so here's the screenshot for history purposes:
image

**Pull request merged**